myfitnesspal/client.py

```diff
@@ -804,3 +810,535 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+            goal["protein"] = protein
+            goal["fat"] = fat
+            # TODO Add micro nutritions
+
+        # Build Post-Request
+        # Post Request
+        url = parse.urljoin(self.BASE_API_URL, "v2/nutrient-goals")
+        result = self.session.post(url, json.dumps(new_goals), headers=auth_header)
+
+        # TODO Check Request Result
+        if result.status_code == 200:
+            return True
+        elif not result.ok:
+            logger.warning(
+                "Request Error - Unable to submit Goals to MyFitnessPal: "
+                "status code: {status}".format(status=result.status_code)
+            )
+            return False
+        else:
+            logger.error(
+                "Request Error - Unable to submit Goals to MyFitnessPal: "
+                "status code: {status}".format(status=result.status_code)
+            )
+            print(result)
+            return False
+
+    def get_recipe_list(self):
+        # TODO EXCEPTION HANDLING
+        recipes_dict = {}
+
+        page_count = 1
+        has_next_page = True
+        while has_next_page:
+            RECIPES_PATH = f"recipe_parser?page={page_count}&sort_order=recent"
+            recipes_url = parse.urljoin(self.BASE_URL_SECURE, RECIPES_PATH)
+            document = self._get_document_for_url(recipes_url)
+            recipes = document.xpath(
+                "//*[@id='main']/ul[1]/li"
+            )  # get all items in the recipe list
+            for recipe_info in recipes:
+                recipe_path = recipe_info.xpath("./div[2]/h2/span[1]/a")[0].attrib[
+                    "href"
+                ]
+                recipe_id = recipe_path.split("/")[-1]
+                recipe_title = recipe_info.xpath("./div[2]/h2/span[1]/a")[0].attrib[
+                    "title"
+                ]
+                recipes_dict[recipe_id] = recipe_title
+
+            # Check for Pagination
+            pagination_links = document.xpath('//*[@id="main"]/ul[2]/a')
+            if pagination_links:
+                if (
+                    page_count == 1
+                ):  # If Pagination exists and it is page 1 there have to be a second, but only one href to the next (obviously none to the previous)
+                    page_count += 1
+                elif (
+                    len(pagination_links) > 1
+                ):  # If there are two links, ont to the previous and one to the next
+                    page_count += 1
+                else:
+                    has_next_page = False  # Only one link means it is the last page
+            else:
+                # Indicator for no recipes
+                has_next_page = False
+                if len(recipes_dict) == 0:
+                    return None
+
+        return recipes_dict
+
+    def get_recipe(self, recipeid: int):
+        recipe_PATH = f"/recipe/view/{recipeid}"
+        recipe_url = parse.urljoin(self.BASE_URL_SECURE, recipe_PATH)
```

Comment:
Forgive me, this is a very nit-y request, but the standard python style would rather have this variable named recipe_path (here & below).

```suggestion
        recipe_path = f"/recipe/view/{recipeid}"
        recipe_url = parse.urljoin(self.BASE_URL_SECURE, recipe_path)
```